src/auto-evo/simulation/PopulationSimulation.cs

```diff
@@ -117,36 +121,127 @@ private static void RunSimulationStep(SimulationConfiguration parameters, List<S
         {
+            _ = random;
+
             // Skip if there aren't any species in this patch
             if (species.Count < 1)
                 return;
 
-            // var biome = patch.Biome;
+            var biome = patch.Biome;
+
+            var sunlightInPatch = biome.Compounds[Sunlight].Dissolved * 10000;
+            var hydrogenSulfideInPatch = biome.Compounds[HydrogenSulfide].Density
+                * biome.Compounds[HydrogenSulfide].Amount * 1000;
+            var glucoseInPatch = biome.Compounds[Glucose].Density
+                * biome.Compounds[Glucose].Amount * 1000;
 
             // TODO: this is where the proper auto-evo algorithm goes
 
-            // Here's a temporary boost when there are few species and penalty
-            // when there are many species
-            bool lowSpecies = species.Count <= Constants.AUTO_EVO_LOW_SPECIES_THRESHOLD;
-            bool highSpecies = species.Count >= Constants.AUTO_EVO_HIGH_SPECIES_THRESHOLD;
+            var speciesEnergies = new Dictionary<MicrobeSpecies, float>(species.Count);
 
-            foreach (var currentSpecies in species)
+            var totalPhotosynthesisScore = 0.0f;
+            var totalChemosynthesisScore = 0.0f;
+            var totalPredationScore = 0.0f;
+
+            // Calculate the total scores of each type in the current patch
+            foreach (MicrobeSpecies currentMicrobeSpecies in species)
             {
-                int currentPopulation = populations.GetPopulationInPatch(currentSpecies, patch);
-                int populationChange = random.Next(
-                    -Constants.AUTO_EVO_RANDOM_POPULATION_CHANGE, Constants.AUTO_EVO_RANDOM_POPULATION_CHANGE + 1);
+                totalPhotosynthesisScore += GetPhotosynthesisScore(currentMicrobeSpecies);
+                totalChemosynthesisScore += GetChemosynthesisScore(currentMicrobeSpecies);
+                totalPredationScore += GetPredationScore(currentMicrobeSpecies);
+            }
+
+            // Avoid division by 0
+            totalPhotosynthesisScore = Math.Max(MathUtils.EPSILON, totalPhotosynthesisScore);
+            totalChemosynthesisScore = Math.Max(MathUtils.EPSILON, totalChemosynthesisScore);
+            totalPredationScore = Math.Max(MathUtils.EPSILON, totalPredationScore);
+
+            // Calculate the share of environmental energy captured by each species
+            var energyAvailableForPredation = 0.0f;
+
+            foreach (MicrobeSpecies currentMicrobeSpecies in species)
+            {
+                var currentSpeciesEnergy = 0.0f;
+
+                currentSpeciesEnergy += sunlightInPatch
+                    * GetPhotosynthesisScore(currentMicrobeSpecies) / totalPhotosynthesisScore;
+
+                currentSpeciesEnergy += hydrogenSulfideInPatch
+                    * GetChemosynthesisScore(currentMicrobeSpecies) / totalChemosynthesisScore;
+
+                currentSpeciesEnergy += glucoseInPatch
+                    * 1 / species.Count;
```

Comment:
I think that the glucose score should be done similarly to the other scores by calculating how much glucose the species can turn into ATP, doesn't make sense if a species that relies on iron gets a ton of glucose score.